airflow/www/templates/airflow/variable_list.html

```diff
@@ -1,45 +1,52 @@
 {#
- Licensed to the Apache Software Foundation (ASF) under one
- or more contributor license agreements.  See the NOTICE file
- distributed with this work for additional information
- regarding copyright ownership.  The ASF licenses this file
- to you under the Apache License, Version 2.0 (the
- "License"); you may not use this file except in compliance
- with the License.  You may obtain a copy of the License at
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
 
-   http://www.apache.org/licenses/LICENSE-2.0
+    http://www.apache.org/licenses/LICENSE-2.0
 
- Unless required by applicable law or agreed to in writing,
- software distributed under the License is distributed on an
- "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
- KIND, either express or implied.  See the License for the
- specific language governing permissions and limitations
- under the License.
-#}
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+ #}
 
-{% extends 'appbuilder/general/model/list.html' %}
+ {% extends 'appbuilder/general/model/list.html' %}
+ {% from 'airflow/_messages.html' import show_message %}
 
-{% block content %}
-  {% if can_create_variable() %}
-  <div class="well well-sm pull-left">
-    <form class="form-inline" action="{{ url_for('VariableModelView.varimport') }}" method="post" enctype="multipart/form-data">
-      {% if csrf_token %}
-        <input type="hidden" name="csrf_token" value="{{ csrf_token() }}"/>
-      {% endif %}
-      <div class="form-group">
-        <input class="form-control-file" type="file" name="file">
-      </div>
-      <button type="submit" class="btn">
-        <span class="material-icons">cloud_upload</span>
-        Import Variables
-      </button>
-    </form>
-  </div>
-  <div class="clearfix"></div>
-  {% endif %}
-  {{ super() }}
-  <style>
-    td { white-space: nowrap; text-overflow: ellipsis; overflow: hidden; max-width:1px;}
-    th { resize: horizontal; overflow: auto;}
-  </style>
-{% endblock %}
+ {% block content %}
+
+   {% if is_using_secrets_backend() %}
+    {% call show_message(category='warning') %}
+      <b>WARNING: </b>The Airflow UI will not display Connections or Variables stored in an alternative secrets backend.
```

Comment:
List those alternative backends distributed with Airflow here too.

And if you want extra credit, maybe count the number of connections/variables not shown, only show the warning if there are  include that number in the warning message:

```jinja
WARNING:
{% trans count=num_variables %}
The Airflow UI will not display the {{ num_variables }} variable stored
in alternative secrets backends (like EnvironmentVariablesBackend
and SecretsBackend).
{% pluralize %}
The Airflow UI will not display the {{ num_variables }} variables stored
in alternative secrets backends (like EnvironmentVariablesBackend
and SecretsBackend).
{% endtrans %}
```